awx/ui-next/Makefile

```diff
@@ -0,0 +1,87 @@
+ui_next_mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+ui_next_dir := $(dir $(ui_next_mkfile_path))
+
+ifeq ($(current_dir), undefined)
+	ui_next_dir := $(dir $(ui_next_mkfile_path))
+else
+	ui_next_dir := $(shell realpath --relative-to $(current_dir) $(ui_next_dir))/
+endif
+
+ANSIBLE_UI_GIT_REPO_HTTPS ?= https://github.com/ansible/ansible-ui.git
+ANSIBLE_UI_GIT_REPO_SSH ?= git@github.com:ansible/ansible-ui.git
+
+ANSIBLE_UI_GIT_BRANCH ?= main
+
+ANSIBLE_UI_LOCAL ?=
+
+ANSIBLE_UI_DIR ?= $(ui_next_dir)ansible-ui
+
+.PHONY: ui-next/clone-https
+## Shallow clone the ui-next repo via https skip if ANSIBLE_UI_GIT_REPO_HTTPS is undefined
+ui-next/clone-https:
```

Comment:
I don't think these necessarily need to be separate rules. I would simplify by making it `ANSIBLE_UI_URL`. If you need to do different things when it's https or git, then you can check the `ANSIBLE_UI_URL` whenever you want. At the very least, I'd probably collapse the rules into a `ui-next/clone` or use $(ANSIBLE_UI_SRCDIR) as a target, and decide how to clone it in that target rule.